### PR TITLE
Adding deprecated course redirects

### LINKED ISSUE #2868: Course deprecation
Some courses have become redundant or outdated with the initial learning pathways now published. To streamline our offerings and ensure we provide the most current content to our learners, @westnz and @jonathanbossenger want to deprecate the following courses and direct users to the relevant learning pathway.

- https://learn.wordpress.org/course/simple-site-design-with-full-site-editing/
- https://learn.wordpress.org/course/part-2-personalized-site-design-with-full-site-editing-and-theme-blocks/
- https://learn.wordpress.org/course/part-3-advanced-site-design-with-full-site-editing-site-editor-templates-and-template-parts/
- https://learn.wordpress.org/course/developing-with-the-wordpress-rest-api/
- https://learn.wordpress.org/course/converting-a-shortcode-to-a-block/
- https://learn.wordpress.org/course/a-developers-guide-to-block-themes-part-1/
- https://learn.wordpress.org/course/a-developers-guide-to-block-themes-part-2/

### PR DESCRIPTION
**Pull request by jonathanbossenger**
Fixes #2868

### CODE REVIEW

**Comment by jonathanbossenger** [comment]
@adamwoodnz I'm not sure what you think of what I did here, but I thought it would make sense to group redirects by type.

File: wp-content/plugins/wporg-learn/inc/redirects.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ function wporg_learn_redirect_old_urls() {
 		return;
 	}
 
-	$redirects = array(
+	$pages = array(
 		// Source => Destination, any characters after the source will be appended to the destination.
```
Comment (adamwoodnz) [inline wp-content/plugins/wporg-learn/inc/redirects.php]: Should probably move this comment out of this block as it applies to all arrays

File: wp-content/plugins/wporg-learn/inc/redirects.php [hunk 2]
```diff
@@ -49,7 +49,7 @@ function wporg_learn_redirect_old_urls() {
 		return;
 	}
 
-	$redirects = array(
+	$pages = array(
```
Comment (adamwoodnz) [inline wp-content/plugins/wporg-learn/inc/redirects.php:55]: Not all of the redirects in this group are pages, some are post types. If we're going to split out courses we should do the same for tutorials.

**Comment by adamwoodnz** [comment]
> @adamwoodnz I'm not sure what you think of what I did here, but I thought it would make sense to group redirects by type.

I'm not opposed to doing this if you think there's value in organising them. Left some comments about going further with the approach if so.

File: wp-content/plugins/wporg-learn/inc/redirects.php (hunk 2 above)
Comment (jonathanbossenger) [reply wp-content/plugins/wporg-learn/inc/redirects.php:55]: I agree; that's why I wanted to check with you first. I'll update the PR today to separate these better.

**Comment by jonathanbossenger** [comment]
Thanks @adamwoodnz PR updated based on your feedback.

**Review by adamwoodnz — APPROVED** [review]
LGTM :shipit:

**Pull request merged**